### PR TITLE
Automations

### PR DESCRIPTION
**Pull request by Enkhtuvshin0513**
### Context

**Segment**
We have improved our segment function. Now users can create more advanced and complex segments. More details below

1. Now users can choose multiple property types in segments. For example, it will allow creating customer segments with a deal filter.

https://user-images.githubusercontent.com/37796969/135966158-92948165-53c3-4868-a5ec-7c74361a30db.mov

2. Added and, or conjunctions in segments properties and event filters.

https://user-images.githubusercontent.com/37796969/135966171-8bd48cfd-fa47-47b6-950d-d11e061c717d.mov

3. Made it possible to add multiple segment groups in the filter.

https://user-images.githubusercontent.com/37796969/135966183-074193e4-1802-4bf5-bdd9-7ac08cdf7137.mov

### CODE REVIEW

File: api/src/cronJobs/activityLogs.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import * as dotenv from 'dotenv';
 import * as schedule from 'node-schedule';
+// import { RABBITMQ_QUEUES } from '../data/constants';
```
Comment (batamar) [inline api/src/cronJobs/activityLogs.ts:3]: ?

File: api/src/cronJobs/activityLogs.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 import { ACTIVITY_LOG_ACTIONS, putActivityLog } from '../data/logUtils';
-import { fetchBySegments } from '../data/modules/segments/queryBuilder';
+import { fetchSegment } from '../data/modules/segments/queryBuilder';
 import { connect } from '../db/connection';
-import { Companies, Customers, Segments } from '../db/models';
+import { Segments } from '../db/models';
+// import messageBroker from '../messageBroker';
```
Comment (batamar) [inline api/src/cronJobs/activityLogs.ts:8]: ?

File: api/src/data/logUtils.ts [hunk 3]
```diff
@@ -1488,6 +1489,11 @@ export const putCreateLog = async (
 
   callAfterMutation({ ...params, action: LOG_ACTIONS.CREATE }, user);
 
+  messageBroker().sendMessage(RABBITMQ_QUEUES.AUTOMATIONS_TRIGGER, {
```
Comment (batamar) [inline api/src/data/logUtils.ts:1492]: do we need this ?

**Review by batamar — COMMENTED** [review]
reset  api/src/data/resolvers/queries/configs.ts changes

File: automations/src/actions/setProperty.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+export type Output = {
+  result: number;
+}
+
+const getToChangeObjects = async ({ triggerType, target, module }) => {
+  if (module === triggerType) {
+    return [target];
+  }
+
+  // if (Object.keys(target).includes(`${module}Id`)) {
```
Comment (batamar) [inline automations/src/actions/setProperty.ts:38]: ?

**Comment by Enkhtuvshin0513** [comment]
### Context
**Segment**

**Comment by Enkhtuvshin0513** [comment]
### Context

**Segment**
We have improved our segment function. Now users can create more advanced and complex segments. More details below

1. Now users can choose multiple property types in segments. For example, it will allow creating customer segments with a deal filter.

https://user-images.githubusercontent.com/37796969/135960692-13f51090-9648-49f3-90b4-62c19cdca47d.mov


2. Added and, or conjunctions in segments properties and event filters.

https://user-images.githubusercontent.com/37796969/135960701-5d12913c-acbf-4001-9b33-fa06f64e1ab4.mov


3. Made it possible to add multiple segment groups in the filter.

https://user-images.githubusercontent.com/37796969/135960714-f76de5e8-ca76-4c57-a8e4-aeb768568916.mov

**Comment by munkhsaikhan** [comment]
### **Context.**
Beta version of automation.

**developed triggers (triggers are segment based):** 

> Contact based, Company based, Conversation based, Task based, Ticket based, Sales pipeline based and Team member based

**developed actions:**

> Branches(segment based), Manage Properties, Create Task, Created Deal, Create Ticket and Delay

**Automations:**
1. Automations list
Screenshot

2. Automation draw detail
allow drag, connect, double click and hover
Screenshot1

3. Show Automation working histories
Screenshot2

4. Automation's trigger detail / configure
Screenshot3

5. Automation's action detail / configure
Screenshot4

**Pull request merged**